### DISCUSSION TITLE
Datatype 'i4' not understood

### DISCUSSION THREAD

**Discussion by WalauA** [dcajasn/Riskfolio-Lib#203] [Q&A] [answered]
When importing riskfolio this error type keeps appearing.
I have upgrade the package itself as well.
Never used github before so here is the error.

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[35], line 1
----> 1 import riskfolio as rp

File ~\anaconda3\Lib\site-packages\riskfolio\__init__.py:9
      1 """"""  #
      2 """
      3 Copyright (c) 2020-2024, Dany Cajas
      4 All rights reserved.
      5 This work is licensed under BSD 3-Clause "New" or "Revised" License.
      6 License available at https://github.com/dcajasn/Riskfolio-Lib/blob/master/LICENSE.txt
      7 """
----> 9 from riskfolio.src import *
     10 from riskfolio.external import *
     11 from riskfolio.version import (
     12     version as __version__,
     13 )

File ~\anaconda3\Lib\site-packages\riskfolio\src\__init__.py:9
      1 """"""  #
      2 """
      3 Copyright (c) 2020-2024, Dany Cajas
      4 All rights reserved.
      5 This work is licensed under BSD 3-Clause "New" or "Revised" License.
      6 License available at https://github.com/dcajasn/Riskfolio-Lib/blob/master/LICENSE.txt
      7 """
----> 9 from .Portfolio import *
     10 from .HCPortfolio import *
     11 from .PlotFunctions import *

File ~\anaconda3\Lib\site-packages\riskfolio\src\Portfolio.py:17
     15 from sklearn.decomposition import PCA
     16 from sklearn.preprocessing import StandardScaler
---> 17 import riskfolio.src.RiskFunctions as rk
     18 import riskfolio.src.ParamsEstimation as pe
     19 import riskfolio.src.AuxFunctions as af

File ~\anaconda3\Lib\site-packages\riskfolio\src\RiskFunctions.py:13
     11 import cvxpy as cp
     12 import riskfolio.src.OwaWeights as owa
---> 13 import riskfolio.src.ParamsEstimation as pe
     14 from scipy.optimize import minimize
     15 from scipy.optimize import Bounds

File ~\anaconda3\Lib\site-packages\riskfolio\src\ParamsEstimation.py:21
     18 from numpy.linalg import inv
     19 from itertools import product
---> 21 import riskfolio.src.AuxFunctions as af
     22 import riskfolio.src.DBHT as db
     23 import riskfolio.src.GerberStatistic as gs

File ~\anaconda3\Lib\site-packages\riskfolio\src\AuxFunctions.py:23
     21 from sklearn.neighbors import KernelDensity
     22 from sklearn.metrics import silhouette_samples
---> 23 from astropy.stats import knuth_bin_width, freedman_bin_width, scott_bin_width
     24 from itertools import product
     25 import riskfolio.external.cppfunctions as cf

File ~\anaconda3\Lib\site-packages\astropy\stats\__init__.py:15
      2 """
      3 This subpackage contains statistical tools provided for or used by Astropy.
      4 
   (...)
     11 
     12 """
     14 from . import bayesian_blocks as _bb
---> 15 from . import (
     16     biweight,
     17     circstats,
     18     funcs,
     19     info_theory,
     20     jackknife,
     21     sigma_clipping,
     22     spatial,
     23 )
     24 from . import histogram as _hist
     25 from .bayesian_blocks import *

File ~\anaconda3\Lib\site-packages\astropy\stats\circstats.py:15
      3 """
      4 This module contains simple functions for dealing with circular statistics, for
      5 instance, mean, variance, standard deviation, correlation coefficient, and so
   (...)
     10 'CircStats' [2]_.
     11 """
     13 import numpy as np
---> 15 from astropy.units import Quantity
     17 __all__ = [
     18     "circmean",
     19     "circstd",
   (...)
     25     "vonmisesmle",
     26 ]
     27 __doctest_requires__ = {"vtest": ["scipy"]}

File ~\anaconda3\Lib\site-packages\astropy\units\__init__.py:13
      1 # Licensed under a 3-clause BSD style license - see LICENSE.rst
      3 """
      4 This subpackage contains classes and functions for defining and converting
      5 between different physical units.
   (...)
     10 code under a BSD license.
     11 """
---> 13 from . import (
     14     astrophys,
     15     cgs,
     16     core,
     17     decorators,
     18     misc,
     19     photometric,
     20     physical,
     21     quantity,
     22     si,
     23     structured,
     24 )
     25 from .astrophys import *
     26 from .cgs import *

File ~\anaconda3\Lib\site-packages\astropy\units\astrophys.py:10
      3 """
      4 This package defines the astrophysics-specific units.  They are also
      5 available in the `astropy.units` namespace.
      6 """
      8 import numpy as np
---> 10 from astropy.constants import si as _si
     12 from . import si
     13 from .core import UnitBase, def_unit, set_enabled_units

File ~\anaconda3\Lib\site-packages\astropy\constants\__init__.py:27
     23 from astropy import units
     25 del units
---> 27 from . import cgs, si
     28 from . import utils as _utils
     29 from .config import codata, iaudata

File ~\anaconda3\Lib\site-packages\astropy\constants\cgs.py:7
      1 # Licensed under a 3-clause BSD style license - see LICENSE.rst
      2 """
      3 Astronomical and physics constants in cgs units.  See :mod:`astropy.constants`
      4 for a complete listing of constants defined in Astropy.
      5 """
----> 7 from .config import codata, iaudata
      8 from .constant import Constant
     10 for _nm, _c in (*sorted(vars(codata).items()), *sorted(vars(iaudata).items())):

File ~\anaconda3\Lib\site-packages\astropy\constants\config.py:15
     12 phys_version = astropy.physical_constants.get()
     13 astro_version = astropy.astronomical_constants.get()
---> 15 codata = importlib.import_module(".constants." + phys_version, "astropy")
     16 iaudata = importlib.import_module(".constants." + astro_version, "astropy")

File ~\anaconda3\Lib\importlib\__init__.py:126, in import_module(name, package)
    124             break
    125         level += 1
--> 126 return _bootstrap._gcd_import(name[level:], package, level)

File ~\anaconda3\Lib\site-packages\astropy\constants\codata2018.py:9
      2 """
      3 Astronomical and physics constants in SI units.  See :mod:`astropy.constants`
      4 for a complete listing of constants defined in Astropy.
      5 """
      7 import math
----> 9 from .constant import Constant, EMConstant
     11 # PHYSICAL CONSTANTS
     12 # https://en.wikipedia.org/wiki/2019_redefinition_of_SI_base_units
     15 class CODATA2018(Constant):

File ~\anaconda3\Lib\site-packages\astropy\constants\constant.py:10
      7 import numpy as np
      9 from astropy.units.core import Unit, UnitsError
---> 10 from astropy.units.quantity import Quantity
     11 from astropy.utils import lazyproperty
     12 from astropy.utils.exceptions import AstropyUserWarning

File ~\anaconda3\Lib\site-packages\astropy\units\quantity.py:38
     28 from .core import (
     29     Unit,
     30     UnitBase,
   (...)
     35     get_current_unit_registry,
     36 )
     37 from .format import Base, Latex
---> 38 from .quantity_helper import can_have_arbitrary_unit, check_output, converters_and_unit
     39 from .quantity_helper.function_helpers import (
     40     DISPATCHED_FUNCTIONS,
     41     FUNCTION_HELPERS,
     42     SUBCLASS_SAFE_FUNCTIONS,
     43     UNSUPPORTED_FUNCTIONS,
     44 )
     45 from .structured import StructuredUnit, _structured_unit_like_dtype

File ~\anaconda3\Lib\site-packages\astropy\units\quantity_helper\__init__.py:16
      8 from .converters import *
     10 # isort: split
     11 # By importing helpers, all the unit conversion functions needed for
     12 # numpy ufuncs and functions are defined.
     13 # For scipy.special and erfa, importing the helper modules ensures
     14 # the definitions are added as modules to UFUNC_HELPERS, to be loaded
     15 # on demand.
---> 16 from . import erfa, function_helpers, helpers, scipy_special

File ~\anaconda3\Lib\site-packages\astropy\units\quantity_helper\erfa.py:5
      2 """Quantity helpers for the ERFA ufuncs."""
      3 # Tests for these are in coordinates, not in units.
----> 5 from erfa import dt_eraASTROM, dt_eraLDBODY, dt_pv
      6 from erfa import ufunc as erfa_ufunc
      8 from astropy.units.core import UnitsError, UnitTypeError, dimensionless_unscaled

File ~\anaconda3\Lib\site-packages\erfa\__init__.py:5
      1 # Licensed under a 3-clause BSD style license - see LICENSE.rst
      2 
      3 # Import version first, as this gives a more useful error message
      4 # if a system liberfa is too old.
----> 5 from .version import version as __version__  # noqa
      6 from .core import *  # noqa
      7 from .ufunc import (dt_eraASTROM, dt_eraLDBODY, dt_eraLEAPSECOND,  # noqa
      8                     dt_pv, dt_sign, dt_type, dt_ymdf, dt_hmsf, dt_dmsf)

File ~\anaconda3\Lib\site-packages\erfa\version.py:25
     22 # Set the version numbers a bit indirectly, so that Sphinx can pick up
     23 # up the docstrings and list the values.
     24 try:
---> 25     from . import ufunc
     26 except ImportError as exc:
     27     # If compiled to use a system liberfa, that library can be too old, and
     28     # miss functions that are available in newer liberfa. If so, we should
     29     # bail since nothing will work, but let's try to give a more informative
     30     # error message.
     31     try:

TypeError: data type 'i4' not understood
​

**Comment by dcajasn** [comment, marked as answer]
Hi @WalauA,

It's not a Riskfolio-Lib problem, it's an astropy problem. Reinstall astropy.

**Reply by WalauA** [reply]
It worked. Thank you kind sir.
